ruby/ql/src/queries/modeling/internal/Summaries.qll

```diff
@@ -0,0 +1,69 @@
+/**
+ * Contains predicates for generating `summaryModel`s to summarize flow through methods.
+ */
+
+private import ruby
+private import codeql.ruby.ApiGraphs
+private import codeql.ruby.TaintTracking
+private import Util as Util
+
+/**
+ * Contains predicates for generating `summaryModel`s to summarize flow through methods.
+ */
+module Summaries {
+  private module Config implements DataFlow::ConfigSig {
```

Comment:
You probably want to use the `FeatureEqualSourceSinkCallContext`, [as in C#](https://github.com/github/codeql/blob/main/csharp/ql/src/utils/modelgenerator/internal/CaptureModels.qll#L163), as that will restrict flow from the source to the sink to be within the same method.

Reply:
Thanks - I had forgotten about this. It removes a few interesting cases, e.g.
```ruby
def foo(a, b)
  res = [a, @b]
  @b = b
  res
end
```
which previously had taint flow from `Argument[1]` to `ReturnValue`. I would consider this an improvement since there isn't really flow from `b` to the return value in a single call, though it's a case that I hadn't considered before.